packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_masternode_create.test.ts

```diff
@@ -0,0 +1,186 @@
+  expect(outs[0].n).toStrictEqual(0)
+  expect(outs[0].tokenId).toStrictEqual(0)
+  expect(outs[0].scriptPubKey.asm.startsWith('OP_RETURN 4466547843')).toBeTruthy()
+  expect(outs[0].scriptPubKey.hex).toStrictEqual(expectedRedeemScript)
+  expect(outs[0].scriptPubKey.type).toStrictEqual('nulldata')
+
+  expect(outs[1].value).toStrictEqual(2)
+  expect(outs[1].n).toStrictEqual(1)
+  expect(outs[1].tokenId).toStrictEqual(0)
+  expect(outs[1].scriptPubKey.type).toStrictEqual('witness_v0_keyhash')
+  expect(outs[1].scriptPubKey.reqSigs).toStrictEqual(1)
+  expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+  const masternodesAfter = await jsonRpc.masternode.listMasternodes()
+  const masternodesAfterLength = Object.keys(masternodesAfter).length
+  expect(masternodesAfterLength).toStrictEqual(masternodesBeforeLength + 1)
+})
+
+it('should be failed if address is P2SH, other than P2PKH AND P2WPKH', async () => {
+  const balance = await container.call('getbalance')
+  expect(balance >= 2).toBeTruthy()
+
+  const address = await container.getNewAddress('', 'p2sh-segwit')
+  const addressDest: P2SH = P2SH.fromAddress(RegTest, address, P2SH)
+  const addressDestKeyHash = addressDest.hex
+
+  const createMasternode: CreateMasterNode = {
+    operatorType: 0x02,
+    operatorAuthAddress: addressDestKeyHash
+  }
+
+  const script = await providers.elliptic.script()
+
+  const txn: TransactionSegWit = await builder.masternode.create(createMasternode, script)
+
+  const promise = sendTransaction(container, txn)
+  await expect(promise).rejects.toThrow(DeFiDRpcError)
+  await expect(promise).rejects.toThrow('CreateMasternodeTx: bad owner and|or operator address (should be P2PKH or P2WPKH only) or node with those addresses exists')
+})
+
+it('should be failed while collateral is not 2', async () => {
+  expect.assertions(2)
+
+  const balanceBefore = await container.call('getbalance')
+
+  await container.call('sendtoaddress', ['bcrt1ql0ys2ahu4e9uhjn2l0mehhh4e0mmh7npyhx0re', balanceBefore - 2])
+
+  const balanceAfter = await container.call('getbalance')
+  console.log('balanceAfter: ', balanceAfter)
+  expect(balanceAfter < 2).toBeTruthy()
+
+  const address = await container.getNewAddress('', 'legacy')
+  const addressDest: P2PKH = P2PKH.fromAddress(RegTest, address, P2PKH)
+  const addressDestHex = addressDest.hex
+
+  const createMasternode: CreateMasterNode = {
+    operatorType: 0x01,
+    operatorAuthAddress: addressDestHex
+  }
+
+  await fundEllipticPair(container, providers.ellipticPair, 1.1)
+
+  const script = await providers.elliptic.script()
+
+  const txn: TransactionSegWit = await builder.masternode.create(createMasternode, script)
+
+  try {
+    await sendTransaction(container, txn)
+  } catch (err) {
+    expect(err.message).toStrictEqual('DeFiDRpcError: \'CreateMasternodeTx: malformed tx vouts (wrong creation fee or collateral amount) (code 16)\', code: -26')
+  }
+})
```

Comment:
need to add timelock tests

Reply:
currently build doesn't support `timelock`

https://github.com/DeFiCh/ain/blob/e6caba0c0b220182b69e6043502a51c3f567c599/src/masternodes/mn_checks.h#L167-L170

for this i might need build another docker image with eunospaya release, right??

Reply:
rebase the `main` branch, it supports mn.timeLock